### PR TITLE
Hotel Search Problem

### PR DESCRIPTION
**Pull request by genintho**

### CODE REVIEW

File: hotel_search/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Building an Hotel Search API
```
Comment (lucap) [inline hotel_search/README.md]: Build a hotel search API

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 *.pyc
 hipproblems.egg-info
+.idea/*
```
Comment (lucap) [inline .gitignore]: what's this from?

File: README.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 This is a collection of (simplified) problems that we face at Hipmunk.
 
 - [Building a Flight Search API](https://github.com/Hipmunk/hipproblems/tree/master/searchrunner)
+- [Building an Hotel Search API](https://github.com/Hipmunk/hipproblems/tree/master/hotel_search)
```
Comment (lucap) [inline README.md]: Build a hotel search API

File: README.md [hunk 4]
```diff
@@ -15,4 +16,4 @@ This is a collection of (simplified) problems that we face at Hipmunk.
 
 If the setup script fails make sure you have [setuptools](https://pypi.python.org/pypi/setuptools) installed and try again.
 
-Note, the work queue requires a [Redis](http://redis.io/) server.
\ No newline at end of file
+Note, the work queue requires a [Redis](http://redis.io/) server.
```
Comment (lucap) [inline README.md]: This installation section is confusing. If redis is only required for one problem, can we add a section in the problem itself. 

Or at least say:
Note: "Implementing a Distributed Work Queue" requires a Redis server.

File: hotel_search/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+When a user does an hotel search on Hipmunk we search many partner sites simultaneously to ensure we give them the best
```
Comment (lucap) [inline hotel_search/README.md]: `When a user runs a hotel search`

File: hotel_search/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+- United
+
+# The Problem
+
+Your job is to build an API that queries each provider and returns a merged list containing all of their results.
+
+Requirements:
+- You must search all providers
+- The results should be sorted by ecstasy
+- The scraper APIs already return results sorted by ecstasy, you should take advantage of this!
```
Comment (lucap) [inline hotel_search/README.md]: is ecstasy still in there or is it agony?

File: hotel_search/scraperapi.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
+from tornado import gen, ioloop, web
+from hotel_search.scrapers import get_scraper
+
+class ScraperApiHandler(web.RequestHandler):
+
+    @gen.coroutine
+    def get(self, provider):
+        scraper_cls = get_scraper(provider)
+        if not scraper_cls:
+            self.set_status(400)
```
Comment (lucap) [inline hotel_search/scraperapi.py]: to be technically correct, we want 404 here.

File: hotel_search/scraperapi_test.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        expected = EXPECTED_COUNTS[provider]
+        assert count == expected, "%s results missing for %s" % (
+            expected - count,
+            provider,
+        )
+
+    sorted_results = sorted(results, key=lambda r: r["agony"])
+    assert results == sorted_results, "Results aren't sorted properly!"
+
+    took = resp.elapsed.total_seconds()
```
Comment (lucap) [inline hotel_search/scraperapi_test.py]: to be fair, we should move that statement before we do the verification.

File: hotel_search/scrapers/common.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+    @gen.coroutine
+    def run(self):
+        self.results = []
+
+        # wait a bit
+        yield gen.sleep(2)
+
+        self.load_results()
+        self.results.sort(key=lambda r: r['ecstasy'])
```
Comment (lucap) [inline hotel_search/scrapers/common.py]: this should be agony

File: hotel_search/scrapers/common.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+        raise gen.Return(self.results)
+
+    def load_results(self):
```
Comment (lucap) [inline hotel_search/scrapers/common.py]: I don't think we need this function, we could just inline it.

File: hotel_search/scraperapi_test.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
+from collections import Counter
+import requests
+
+
+EXPECTED_COUNTS = {
+    "Expedia": 1199,
+    "Orbitz": 300,
+    "Priceline": 600,
+    "Travelocity": 400,
+    "United": 1799,
```
Comment (lucap) [inline hotel_search/scraperapi_test.py]: are these numbers going to be correct if we do `self.load_fake_results(xrange(1, 20, self.step))` later ?

File: .gitignore (hunk 2 above)
Comment (lucap) [inline .gitignore]: Intellij IDE cool.

File: hotel_search/scraperapi.py (hunk 7 above)
Comment (genintho) [inline hotel_search/scraperapi.py]: I think you mean semantically ;)

File: hotel_search/scrapers/common.py (hunk 10 above)
Comment (genintho) [inline hotel_search/scrapers/common.py]: @lucap I am copying the structure of the search_runner.

File: hotel_search/README.md (hunk 6 above)
Comment (genintho) [inline hotel_search/README.md]: I use ecstasy in the Hotel challenge code.

**Pull request merged**